deid/bene_id_mapping.sql

```diff
@@ -78,25 +84,149 @@ from (
 
 
+-- First, insert people without bene_ids (who therefore need a date shift)
+insert /*+ APPEND */ into msis_person
+select
+  umid.bene_id bene_id,
+  umid.msis_id msis_id,
+  umid.state_cd state_cd,
+  CASE 
+    WHEN prev_msis.msis_id is not null and prev_msis.state_cd is not null
+    THEN prev_msis.date_shift_days
+    ELSE round(dbms_random.value(-364,0))
+  END date_shift_days,
+  CASE 
+    WHEN prev_msis.msis_id is not null and prev_msis.state_cd is not null
+    THEN prev_msis.dob_shift_months
+    ELSE dob_shift.dob_shift_months
+  END dob_shift_months
+from (
+  -- The Personal Summary File contains one record for every individual enrolled 
+  -- for at least one day during the year.  So, it's not actually necessary to
+  -- select from the union of all the Medicaid tables with the production data.
+  -- However, in order for the process to work for the sample data, we need to
+  -- look at all the tables.
+  -- https://www.resdac.org/cms-data/files/max-ps
+  select /*+ PARALLEL(MAXDATA_PS,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_PS where bene_id is null
+  union
+  select /*+ PARALLEL(MAXDATA_IP,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_IP where bene_id is null
+  union
+  select /*+ PARALLEL(MAXDATA_LT,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_LT where bene_id is null
+  union
+  select /*+ PARALLEL(MAXDATA_OT,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_OT where bene_id is null
+  union
+  select /*+ PARALLEL(MAXDATA_RX,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_RX where bene_id is null
+  ) umid
+left join dob_shift on dob_shift.msis_id = umid.msis_id 
+  and dob_shift.state_cd = umid.state_cd
+left join "&&prev_cms_id_schema"."&&msis_person_prev_yrs_cumu" prev_msis 
+  on prev_msis.msis_id = umid.msis_id and prev_msis.state_cd = umid.state_cd
+;
+commit;
+-- Next, get everyone who has a bene_id (who will get a date/dob shift per bene_id)
+insert /*+ APPEND */ into msis_person
+select
+  umid.bene_id bene_id,
+  umid.msis_id msis_id,
+  umid.state_cd state_cd,
+  null date_shift_days,
+  null dob_shift_months
+from (
+  select /*+ PARALLEL(MAXDATA_PS,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_PS where bene_id is not null
+  union
+  select /*+ PARALLEL(MAXDATA_IP,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_IP where bene_id is not null
+  union
+  select /*+ PARALLEL(MAXDATA_LT,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_LT where bene_id is not null
+  union
+  select /*+ PARALLEL(MAXDATA_OT,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_OT where bene_id is not null
+  union
+  select /*+ PARALLEL(MAXDATA_RX,12) */ distinct bene_id, msis_id, state_cd from MAXDATA_RX where bene_id is not null
+  ) umid;
+commit;
+create unique index msis_person_mid_st_idx on msis_person (msis_id, state_cd);
+
+
+------------------------------------------------------------------------------------------

```

Comment:
not cricitcal, but I'm not a fan of ASCII-art comments. And without some text telling me what the line separates, I'm not sure what it's there for.

My preference is JavaDoc-style comment blocks above things to say what they are:

```
/**
 * Bubblegum - a yummy chewey thing.
 */
```

Reply:
removed ASCII-art comment.